qt/widgets/common/src/DataProcessorUI/QDataProcessorWidget.cpp

```diff
@@ -306,31 +308,167 @@ void QDataProcessorWidget::selectAll() { ui.viewTable->selectAll(); }
+  enableAction(DataProcessorAction::PAUSE);
+}
+
+void QDataProcessorWidget::enableActionOnWidget(QWidget &widget, int index) {
+  enable(*(widget.actions()[index]));
+}
 
-  // Disable 'pause' buttons
-  ui.rowToolBar->actions()[1]->setEnabled(false);
-  m_contextMenu->actions()[1]->setEnabled(false);
+void QDataProcessorWidget::enableActionOnToolbar(DataProcessorAction toEnable) {
+  enableActionOnWidget(*ui.rowToolBar, toToolbarIndex(toEnable));
+}
+
+int QDataProcessorWidget::toToolbarIndex(DataProcessorAction action) {
+  return toCommandIndex(action);
+}
+
+void QDataProcessorWidget::enable(QAction &toEnable) {
+  toEnable.setEnabled(true);
+}
+
+void QDataProcessorWidget::enable(QWidget &toEnable) {
+  toEnable.setEnabled(true);
 }
 
 /**
 Handle interface when data reduction resumed
 */
 void QDataProcessorWidget::resume() {
+  disableResumeButtons();
+  preventTableModification();
+  enablePauseButtons();
+}
+
+void QDataProcessorWidget::preventTableModification() {
+  disableTableModification([this](DataProcessorAction action)
+                               -> void { this->disableAction(action); });
+  disableSelectionAndEditing();
+}
 
-  // Enable 'resume' buttons
-  ui.rowToolBar->actions()[0]->setEnabled(false);
-  m_contextMenu->actions()[0]->setEnabled(false);
-  ui.buttonProcess->setEnabled(false);
+void QDataProcessorWidget::disableSelectionAndEditing() {
+  ui.viewTable->clearSelection();
+  ui.viewTable->setSelectionMode(QAbstractItemView::SelectionMode::NoSelection);
+  ui.viewTable->setEditTriggers(QAbstractItemView::NoEditTriggers);
+}
+
+void QDataProcessorWidget::enableSelectionAndEditing() {
+  ui.viewTable->setSelectionMode(
+      QAbstractItemView::SelectionMode::ContiguousSelection);
+  ui.viewTable->setEditTriggers(QAbstractItemView::AllEditTriggers);
+}
+
+void QDataProcessorWidget::enableResumeButtons() {
+  enableAction(DataProcessorAction::PROCESS);
+  enable(*ui.buttonProcess);
+}
+
+void QDataProcessorWidget::disableResumeButtons() {
+  disableAction(DataProcessorAction::PROCESS);
+  disable(*ui.buttonProcess);
+}
+
+void QDataProcessorWidget::enableAction(DataProcessorAction toEnable) {
+  enableActionOnToolbar(toEnable);
+  enableActionOnContextMenu(toEnable);
+}
+
+void QDataProcessorWidget::disableAction(DataProcessorAction toDisable) {
+  disableActionOnToolbar(toDisable);
+  disableActionOnContextMenu(toDisable);
+}
+
+void QDataProcessorWidget::disableActionOnContextMenu(
+    DataProcessorAction toDisable) {
+  disableActionOnWidget(*m_contextMenu, toContextMenuIndex(toDisable));
+}
+
+void QDataProcessorWidget::enableActionOnContextMenu(
+    DataProcessorAction toEnable) {
+  enableActionOnWidget(*m_contextMenu, toContextMenuIndex(toEnable));
+}
+
+int QDataProcessorWidget::toContextMenuIndex(DataProcessorAction action) {
+  return toCommandIndex(action);
+}
+
+int QDataProcessorWidget::toCommandIndex(DataProcessorAction action) {
+  switch (action) {
```

Comment:
Will this be an issue when you are dealing with non-grouped reduction scenarios?